packages/flutter/test/material/refresh_indicator_test.dart

```diff
@@ -378,4 +379,41 @@ void main() {
     expect(refreshCalled, true);
     expect(tester.takeException(), isFlutterError);
   });
+
+  testWidgets('Refresh starts while scroll view moves back to 0.0 after overscroll on iOS', (WidgetTester tester) async {
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+    refreshCalled = false;
+    double lastScrollOffset;
+    final ScrollController controller = new ScrollController();
+
+    await tester.pumpWidget(
+      new MaterialApp(
+        home: new RefreshIndicator(
+          onRefresh: refresh,
+          child: new ListView(
+            controller: controller,
+            physics: const AlwaysScrollableScrollPhysics(),
+            children: <String>['A', 'B', 'C', 'D', 'E', 'F'].map((String item) {
+              return new SizedBox(
+                height: 200.0,
+                child: new Text(item),
+              );
+            }).toList(),
+          ),
+        ),
+      ),
+    );
+
+    await tester.fling(find.text('A'), const Offset(0.0, 300.0), 1000.0);
+    await tester.pump(const Duration(milliseconds: 100));
+    expect(lastScrollOffset = controller.offset, lessThan(0.0));
+    expect(refreshCalled, isFalse);
+
+    await tester.pump(const Duration(milliseconds: 100));
+    expect(controller.offset, greaterThan(lastScrollOffset));
+    expect(controller.offset, lessThan(0.0));
+    expect(refreshCalled, isTrue);
+
+    debugDefaultTargetPlatformOverride = null;
+  });
```

Comment:
shouldn't we also check that it doesn't happen on android, if this is an iOS feature?

Reply:
I believe that's covered by tests for Scrollables/ListViews to ensure that they don't overscroll on Android.